[V8] Upgrade jquery-form from 2.87 to 4.3.0

mlocati opened the pull request:
I checked [all the changes between the two versions](https://github.com/jquery-form/form/compare/a2d9ba99aef9ccb3b0e867fb1716e4ed4f692e53...v4.3.0), and there shouldn't be breaking changes, just bug fixes.
In addition, [our fixes](https://github.com/concrete5/concrete5/commits/8.5.x/concrete/js/build/vendor/jquery-form/jquery-form.js) are already included in version 4.3.0:
| concrete5 | jquery-form |
|---|---|
| f03dc55477b7cf9ad49b19a4cf6093cccdacbef4 | https://github.com/jquery-form/form/commit/844fe7637b2ab97c4d0ae0df4729720e7c394a27 |
| f71a005d2b85d1295330f0f2d8a1db8c8e3cf627 | https://github.com/jquery-form/form/commit/c261723f407cdc31b158d025df7b3af620a28d0f |
| ca5180f6463a61b8c65545119127f2531f0d0b16 | https://github.com/jquery-form/form/commit/c8f9a22a4dd30b49823a4f9531e94680b3eb6e7b |

aembler commented:
Is there a benefit to upgrading? All I see is potential risk.

JohntheFish commented:
> Is there a benefit to upgrading? All I see is potential risk.

You could say the same about upgrading anything and get a bunch of generic replies. You could say the same about upgrading concrete5.

mlocati commented:
Well, the only relevant changes are:

- https://github.com/jquery-form/form/commit/254b2213c0b017ed2aed41fe98f00fd4d87c7654: forms with files are uploaded via ajax on modern browsers, and disabled file inputs are not considered (in version 2.90)
- https://github.com/jquery-form/form/commit/2f7192d133bab9c45f9002509f91a3fa7c9e30ed: fixed specifying "extra data" in addition to form inputs (in version 2.93)
- https://github.com/jquery-form/form/commit/19cd7dd6f889f217d32eddf6bd883041f0cdbe66: added support for reporting progress while sending forms - much needed when sending files (in version 3.00)
- https://github.com/jquery-form/form/commit/34fbed784342a2047668c30bd6d9828d1dcc4786: added support for elements outside <form>, using the form attribute (in version 3.49)
- https://github.com/jquery-form/form/commit/3fe23f55b9481d5feac453f461d1629b3eff33c0: added support for filtering elements to be sent (in version 4.0.0)
- https://github.com/jquery-form/form/commit/01d26735cbad40a304a551270ea721dfd732a090: added support for multiple "success" callbacks (in version 4.0.0)

Other changes are just bug or coding style fixes... I really don't see any breaking change...

aembler commented:
@JohntheFish Sure but this is a major upgrade (2->4) being proposed for a point release in the last stable version of concrete5. We're not doing another 8.5 release after 8.5.5 – so why risk it?

mlocati commented:
They don't respect semver. BTW feel free to close this PR if you wish

aembler commented:
Thanks @mlocati  – since we're bringing in the updated jQuery Form library as part of version 9, and version 9 is scheduled to launch in 2021, I think we can close this.
